tutorials/abap-environment-external-api/abap-environment-external-api.md

```diff
@@ -8,156 +10,244 @@ author_name: Julie Plummer
 
-### Check, save, and activate
+            CATCH cx_root INTO DATA(lx_exception).
+              out->write( lx_exception->get_text( ) ).
+          ENDTRY.
 
-1. Check your syntax (`Ctrl+F2`).
-2. Save (`Ctrl+S`) and activate (`Ctrl+F3`) your class.
+          " Create structures etc
+          TYPES:
+            BEGIN OF ts_tag,
+              property TYPE string,
+              value    TYPE string,
+            END OF ts_tag,
 
+            BEGIN OF ts_node,
+              id        TYPE string,
+              latitude  TYPE string,
+              longitude TYPE string,
+              tags      TYPE STANDARD TABLE OF ts_tag WITH EMPTY KEY,
+            END OF ts_node,
 
-### Run class in ABAP Console
+            BEGIN OF ts_metadata,
+              osm_base  TYPE string,
+              osm_areas TYPE string,
+            END OF ts_metadata,
+
+            BEGIN OF ts_osm,
+              api_version   TYPE string,
+              api_generator TYPE string,
+              note          TYPE string,
+              metadata      TYPE ts_metadata,
+              nodes         TYPE STANDARD TABLE OF ts_node WITH EMPTY KEY,
+            END OF ts_osm.
+
+          DATA ls_osm TYPE ts_osm.
 
-Run your class in the console (`F9`).
 
-The output should look something like this:
-.
-![Image depicting step-9-console](step-9-console.png)
+          TRY.
+            " transform response; store transformed response in structure osm
+            CALL TRANSFORMATION zmdc_osm_xml_st
+            SOURCE XML lv_xml_results
+            RESULT osm = ls_osm.
+
+
+              out->write( | Names of beergardens in Heidelberg (Germany) found on OpenStreetMaps. | ).
+              out->write( | Data is obtained from the OSM API in XML format and read into ABAP using a simple transformation (ST). | ).
+              out->write( | Note: Not all locations have an associated name. | ).
+              out->write( | Generator: { ls_osm-api_generator } | ).
+              out->write( | ---------------------------------- | ).
+
+              " Output data to console
+              DATA lv_index TYPE int4.
+              LOOP AT ls_osm-nodes ASSIGNING FIELD-SYMBOL(<node>).
+                lv_index = sy-tabix.
+                LOOP AT <node>-tags ASSIGNING FIELD-SYMBOL(<tag>).
+                  IF <tag>-property = 'name'.
+                    out->write( | Beergarden number { lv_index }: { <tag>-value } | ).
+                  ENDIF.
+                ENDLOOP.
+              ENDLOOP.
+              out->write( | ---------------------------------- | ).
+
+          " catch any error: In case XSLT is not correct
+              CATCH cx_root INTO DATA(lx_root).
+                out->write( lx_root->get_text( ) ).
+          ENDTRY.
+
+      ENDMETHOD.
+
+    ```
+
+2. Check your syntax (`Ctrl+F2`).
+
+3. Format (`Ctrl+F1`), save (`Ctrl+S`) and activate (`Ctrl+F3`) your class.
+
+
+### Run class in ABAP Console
```

Comment:
Shouldn't it be "console" with a lower case C or is ABAP Console a product name?